src/caffe/layers/cudnn_conv_layer.cpp

```diff
@@ -90,6 +90,8 @@ void CuDNNConvolutionLayer<Dtype>::LayerSetUp(
   use_reshape_ = true;
   // When true, cached bottom and conv descriptors need to be set.
   initialized_cached_descs_ = false;
+  // In case of reusing it
+  WORKSPACE.release();
```

Comment:
I've set the initial size to 4MB. Now it's cache-able. No performance issues. :)